### PR TITLE
first pass at collection level options

### LINKED ISSUE #366: Add per collection options
Allow an `_options` key in collections to enable things like:

-  'flatmapping':  preserving subdirs for modules (also update galaxy.yml to enable feature)
- reponame: override deafult destination repo name
- version: version to set in galaxy.yml for the collection

### PR DESCRIPTION
**Pull request by jctanner**
fixes #366

### CODE REVIEW

File: migrate.py [hunk 1]
```diff
@@ -1331,7 +1340,7 @@ def assemble_collections(checkout_path, spec, args, target_github_org):
                         if os.path.exists(init_py_path):
                             remove(init_py_path, namespace, collection)
 
-                    do_preserve_subdirs = ((args.preserve_module_subdirs and plugin_type == 'modules') or plugin_type in ALWAYS_PRESERVE_SUBDIRS)
+                    do_preserve_subdirs = (((args.preserve_module_subdirs or options.get('preserve_module_subdirs')) and plugin_type == 'modules') or plugin_type in ALWAYS_PRESERVE_SUBDIRS)
```
Comment (webknjaz) [inline migrate.py]: This conditional has too much nesting in one line. Mind multi-lining it?

File: migrate.py [hunk 2]
```diff
@@ -552,6 +555,11 @@ def rewrite_unit_tests_patch(mod_fst, collection, spec, namespace, args):
         )
     )
 
+    preserve_module_subdirs = False
+    options = spec[namespace][collection].get('_options')
```
Comment (mkrizek) [inline migrate.py]: Shouldn't this be `options = spec[namespace][collection].get(COLLECTION_OPTIONS_KEY, {})`? Same below.

File: migrate.py [hunk 3]
```diff
@@ -552,6 +555,11 @@ def rewrite_unit_tests_patch(mod_fst, collection, spec, namespace, args):
         )
     )
 
+    preserve_module_subdirs = False
+    options = spec[namespace][collection].get(COLLECTION_OPTIONS_KEY)
```
Comment (bcoca) [inline migrate.py]: `, {}` to avoid errors

**Comment by jctanner** [comment]
@bcoca what entry do I need to add to the galaxy.yml to make this work?

@mkrizek what should I check to make sure the result is correct in the collections created?

**Comment by mkrizek** [comment]
> @bcoca what entry do I need to add to the galaxy.yml to make this work?

He mentioned `type: flatmap`.

> @mkrizek what should I check to make sure the result is correct in the collections created?

I would check:
1. subdirs are (not) preserved
2. python imports in other plugins and unittests contain preserved (or not preserved) paths - so `import ansible_collections.community.amazon.plugins.modules.cloud.amazon.ec2_vpc_nat_gateway` vs `import ansible_collections.community.amazon.plugins.modules.ec2_vpc_nat_gateway`
3. same as above but for `@patch` in unittests
4. run sanity and unittests in collections and look for import errors

**Comment by jctanner** [comment]
https://github.com/ansible/ansible/issues/66635#issuecomment-577790180

**Comment by jctanner** [comment]
superseded by https://github.com/ansible-community/collection_migration/pull/425

**Pull request closed without merging**